### PR TITLE
Added Fast404 check for drupal 8

### PR DESCRIPTION
**Pull request by shivanshuag**
I Was playing around with the 'project' and found this easy to port.

### CODE REVIEW

File: Check/BestPractices/Fast404.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  public function getAction() {
+    if ($this->score == SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_WARN) {
+      return dt('See https://drupal.org/node/1313592 and default.settings.php for details on how to implement.');
+    }
+  }
+  /**
+   * Implements \SiteAudit\Check\Abstract\calculateScore().
+   */
+  public function calculateScore() {
+    global $conf;
```
Comment (fluxsauce) [inline Check/BestPractices/Fast404.php]: Not needed, remove.

File: Check/BestPractices/Fast404.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+   */
+  public function getResultWarn() {
+    return dt('Fast 404 pages are not enabled.');
+  }
+  /**
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {
+    if ($this->score == SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_WARN) {
+      return dt('See https://drupal.org/node/1313592 and default.settings.php for details on how to implement.');
```
Comment (fluxsauce) [inline Check/BestPractices/Fast404.php]: That's for Drupal 7; what's the Drupal 8 equivalent? https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21EventSubscriber%21Fast404ExceptionHtmlSubscriber.php/8 is the API page, is there a manual?

**Comment by fluxsauce** [comment]
Thank you! A couple minor tweaks are needed.

File: Check/BestPractices/Fast404.php (hunk 2 above)
Comment (shivanshuag) [inline Check/BestPractices/Fast404.php]: There is no documentation for this in d8. I think the d7 guide applies to d8 too and defaults.settings.php also has instructions. But there is no way to disable/ change the setting for fast 404 in drupal 8

File: Check/BestPractices/Fast404.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+  }
+  /**
+   * Implements \SiteAudit\Check\Abstract\calculateScore().
+   */
+  public function calculateScore() {
+    if (\Drupal::config('system.performance')->get('fast_404.enabled')) {
```
Comment (fluxsauce) [inline Check/BestPractices/Fast404.php]: https://github.com/drupal/drupal/blob/8.0.x/sites/default/default.settings.php#L594

How about just make sure `\Drupal::config('system.performance')->get('fast_404.paths')` is not empty? If you want to be more comprehensive, check for exclude_paths / paths / html?

File: Check/BestPractices/Fast404.php (hunk 2 above)
Comment (fluxsauce) [inline Check/BestPractices/Fast404.php]: Sorry for the delay in response, I've been traveling (and still am).

https://github.com/drupal/drupal/blob/8.0.0-beta7/sites/default/default.settings.php#L572 actually has some pretty detailed instructions, linking to that line would be best. We need to make sure that we're not linking to unrelated instructions.

**Comment by shivanshuag** [comment]
I think this is ready to be merged?

File: Check/BestPractices/Fast404.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+   */
+  public function getResultWarn() {
+    return dt('Fast 404 pages are not enabled for any path.');
+  }
+  /**
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {
+    if ($this->score == SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_WARN) {
+      return dt('See https://github.com/drupal/drupal/blob/8.0.x/sites/default/default.settings.php#L594 for details on how to implement.');
```
Comment (fluxsauce) [inline Check/BestPractices/Fast404.php]: Line 572 would be better - https://github.com/drupal/drupal/blob/8.0.x/sites/default/default.settings.php#L572

**Comment by shivanshuag** [comment]
I have changed it

**Comment by fluxsauce** [comment]
Looks good!

**Comment by fluxsauce** [comment]
Thank you, I really appreciate it!

**Pull request merged**